### PR TITLE
Bomb the Boss

### PR DESCRIPTION
**Pull request by Core0verload**
This PR is an inverse of #18476. Instead of making bombs harmless to megafauna, I make them deal some real damage to them. You can now use ghetto gibtonite charges or real toxins bombs in boss fights, for fun and profit. Just don't explode yourself.

Gib radius: 600 damage
Heavy radius: 300 damage
Light radius: 150 damage

Keep in mind that most of the bosses have 2500 HP, and luring the boss on bomb and not getting caught in blast at the same time is hard.

**Token:** #18466

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm [hunk 1]
```diff
@@ -57,8 +57,6 @@
 				charging = 0
 
 /mob/living/simple_animal/hostile/megafauna/legion/death()
-	if(health > 0)
```
Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: why were these removed?

Comment (Core0verload) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: Because the only way to call this with `health > 0` is wizard mode shenanigans. Or badmin stuff. Making it immune to it makes little sense.

And for Legion, one death is nothing.

**Comment by Cheridan** [comment]
Make sure to test this. Since they're multi-tile mobs, do they take damage multiple times as each tile gets ex_act()'d?

**Comment by Core0verload** [comment]
@Cheridan, I tested it on Legion and ash drake. Both took only 600 damage from (14, 0, 0) explosion. So no, damage is dealt only once.

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: The entire point of it being there is to stop wand of death

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 2]
```diff
@@ -50,4 +35,14 @@
...
-		L.gib()
\ No newline at end of file
+		L.gib()
+
+/mob/living/simple_animal/hostile/megafauna/ex_act(severity, target)
+	switch(severity)
+		if(1)
+			adjustBruteLoss(600)
+		if(2)
+			adjustBruteLoss(300)
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: These values may be a bit high. They'd be fine if it was just gibtonite and ttvs such but throwing a couple mini bombs is not hard

Comment (Core0verload) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Minibombs are traitor gear. IEDs are weak and hard to aim. It's usually more simple to get three more shots with KA than to bother with grenades at all.

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: Why? If you wand of death Legion, it just splits in two, and you no longer have your WoD.

**Comment by Jazaen** [comment]
For the record, I tried using toxins bombs, and lobbing darn things with timer is hard, and will damage everything around the boss (not that it matters with colossus), while remote bombs sometimes bug out.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 3]
```diff
@@ -50,4 +35,14 @@
...
+		if(3)
+			adjustBruteLoss(150)
+	updatehealth()
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: not necessary

Comment (Core0verload) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Are you sure? That's how simple animals code up above does it.

**Comment by bgobandit** [comment]
>an actual use of bombs for mining

:100: :100: :100:

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: yes but that's one of three bosses and the only one that isn't affected

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 3 above)
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: adjustBruteLoss() calls AdjustHealth() which calls updatehealth(). Remove the one above this as well then.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Fine I'll just make my own PR

File: code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm]: Easier to just make my own PR rather than argue about this

**Comment by kevinz000** [comment]
rude kor
